File: binding/src/libqcommandlinkbutton.cpp
#include <QCommandLinkButton>
#include <QEvent>
#include <QPaintEvent>
#include <QPushButton>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyleOptionButton>
#include <QWidget>
#include <qcommandlinkbutton.h>
#include "libqcommandlinkbutton.h"
#include "libqcommandlinkbutton.hxx"

QCommandLinkButton* QCommandLinkButton_new(QWidget* parent) {
	 return new VirtualQCommandLinkButton(parent);
}

QCommandLinkButton* QCommandLinkButton_new2() {
	 return new VirtualQCommandLinkButton();
}

QCommandLinkButton* QCommandLinkButton_new3(const libqt_string text) {
	 return new VirtualQCommandLinkButton(QString::fromUtf8(text.data, text.len));
}

QCommandLinkButton* QCommandLinkButton_new4(const libqt_string text, const libqt_string description) {
	 return new VirtualQCommandLinkButton(QString::fromUtf8(text.data, text.len), QString::fromUtf8(description.data, description.len));
}

QCommandLinkButton* QCommandLinkButton_new5(const libqt_string text, QWidget* parent) {
	 return new VirtualQCommandLinkButton(QString::fromUtf8(text.data, text.len), parent);
}

QCommandLinkButton* QCommandLinkButton_new6(const libqt_string text, const libqt_string description, QWidget* parent) {
	 return new VirtualQCommandLinkButton(QString::fromUtf8(text.data, text.len), QString::fromUtf8(description.data, description.len), parent);
}

libqt_string QCommandLinkButton_Tr(const char* s) {
	QString _ret = QCommandLinkButton::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCommandLinkButton_Description(const QCommandLinkButton* self) {
	QString _ret = self->description();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QCommandLinkButton_SetDescription(QCommandLinkButton* self, const libqt_string description) {
	self->setDescription(QString::fromUtf8(description.data, description.len));
}

QSize* QCommandLinkButton_SizeHint(const QCommandLinkButton* self) {
	return new QSize(self->sizeHint());
}

int QCommandLinkButton_HeightForWidth(const QCommandLinkButton* self, int param1) {
	return self->heightForWidth(param1);
}

QSize* QCommandLinkButton_MinimumSizeHint(const QCommandLinkButton* self) {
	return new QSize(self->minimumSizeHint());
}

void QCommandLinkButton_InitStyleOption(const QCommandLinkButton* self, QStyleOptionButton* option) {
	self->initStyleOption(option);
}

libqt_string QCommandLinkButton_Tr2(const char* s, const char* c) {
	QString _ret = QCommandLinkButton::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCommandLinkButton_Tr3(const char* s, const char* c, int n) {
	QString _ret = QCommandLinkButton::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QSize* QCommandLinkButton_QBaseSizeHint(const QCommandLinkButton* self) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_SizeHint_IsBase(true);
	return new QSize(vqcommandlinkbutton->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnSizeHint(const QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_SizeHint_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_SizeHint_Callback>(slot));
}
}

// Base class handler implementation
int QCommandLinkButton_QBaseHeightForWidth(const QCommandLinkButton* self, int param1) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_HeightForWidth_IsBase(true);
	return vqcommandlinkbutton->heightForWidth(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnHeightForWidth(const QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_HeightForWidth_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_HeightForWidth_Callback>(slot));
}
}

// Base class handler implementation
QSize* QCommandLinkButton_QBaseMinimumSizeHint(const QCommandLinkButton* self) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_MinimumSizeHint_IsBase(true);
	return new QSize(vqcommandlinkbutton->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnMinimumSizeHint(const QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_MinimumSizeHint_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_MinimumSizeHint_Callback>(slot));
}
}

// Base class handler implementation
void QCommandLinkButton_QBaseInitStyleOption(const QCommandLinkButton* self, QStyleOptionButton* option) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_InitStyleOption_IsBase(true);
	vqcommandlinkbutton->initStyleOption(option);
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnInitStyleOption(const QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<const VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_InitStyleOption_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_InitStyleOption_Callback>(slot));
}
}

// Derived class handler implementation
bool QCommandLinkButton_Event(QCommandLinkButton* self, QEvent* e) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
	return vqcommandlinkbutton->event(e);
	} else {
	return self->QCommandLinkButton::event(e);
}
}

// Base class handler implementation
bool QCommandLinkButton_QBaseEvent(QCommandLinkButton* self, QEvent* e) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_Event_IsBase(true);
	return vqcommandlinkbutton->event(e);
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnEvent(QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_Event_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_Event_Callback>(slot));
}
}

// Derived class handler implementation
void QCommandLinkButton_PaintEvent(QCommandLinkButton* self, QPaintEvent* param1) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
	vqcommandlinkbutton->paintEvent(param1);
	} else {
	self->QCommandLinkButton::paintEvent(param1);
}
}

// Base class handler implementation
void QCommandLinkButton_QBasePaintEvent(QCommandLinkButton* self, QPaintEvent* param1) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_PaintEvent_IsBase(true);
	vqcommandlinkbutton->paintEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QCommandLinkButton_OnPaintEvent(QCommandLinkButton* self, intptr_t slot) {
	auto* vqcommandlinkbutton = dynamic_cast<VirtualQCommandLinkButton*>(self);
	if (vqcommandlinkbutton && vqcommandlinkbutton->isVirtualQCommandLinkButton) {
vqcommandlinkbutton->setQCommandLinkButton_PaintEvent_Callback(reinterpret_cast<VirtualQCommandLinkButton::QCommandLinkButton_PaintEvent_Callback>(slot));
}
}

void QCommandLinkButton_Delete(QCommandLinkButton* self) {
    delete self;
}

File: binding/src/libqcommandlinkbutton.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcommandlinkbutton.html
pub const qcommandlinkbutton = struct {

    /// New constructs a new QCommandLinkButton object.
    pub fn New(parent: ?*anyopaque) QtC.QCommandLinkButton {
        return qtc.QCommandLinkButton_new(@ptrCast(parent));
    }


    /// New2 constructs a new QCommandLinkButton object.
    pub fn New2() QtC.QCommandLinkButton {
        return qtc.QCommandLinkButton_new2();
    }


    /// New3 constructs a new QCommandLinkButton object.
    pub fn New3(text: []const u8) QtC.QCommandLinkButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QCommandLinkButton_new3(text_str);
    }


    /// New4 constructs a new QCommandLinkButton object.
    pub fn New4(text: []const u8, description: []const u8) QtC.QCommandLinkButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
const description_str = qtc.libqt_string{
    .len = description.len,
    .data = description.ptr,
};
return qtc.QCommandLinkButton_new4(text_str, description_str);
    }


    /// New5 constructs a new QCommandLinkButton object.
    pub fn New5(text: []const u8, parent: ?*anyopaque) QtC.QCommandLinkButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QCommandLinkButton_new5(text_str, @ptrCast(parent));
    }


    /// New6 constructs a new QCommandLinkButton object.
    pub fn New6(text: []const u8, description: []const u8, parent: ?*anyopaque) QtC.QCommandLinkButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
const description_str = qtc.libqt_string{
    .len = description.len,
    .data = description.ptr,
};
return qtc.QCommandLinkButton_new6(text_str, description_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCommandLinkButton_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcommandlinkbutton.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Description(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCommandLinkButton_Description(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcommandlinkbutton.Description: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetDescription(self: ?*anyopaque, description: []const u8) void {
        const description_str = qtc.libqt_string{
    .len = description.len,
    .data = description.ptr,
};
qtc.QCommandLinkButton_SetDescription(@ptrCast(self), description_str);
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCommandLinkButton_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QCommandLinkButton_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCommandLinkButton_QBaseSizeHint(@ptrCast(self));
    }

    pub fn HeightForWidth(self: ?*anyopaque, param1: i32) i32 {
        return qtc.QCommandLinkButton_HeightForWidth(@ptrCast(self), param1);
    }

    /// Allows for overriding the related default method
    pub fn OnHeightForWidth(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) i32) void {
        qtc.QCommandLinkButton_OnHeightForWidth(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseHeightForWidth(self: ?*anyopaque, param1: i32) i32 {
        return qtc.QCommandLinkButton_QBaseHeightForWidth(@ptrCast(self), param1);
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCommandLinkButton_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QCommandLinkButton_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QCommandLinkButton_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn InitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QCommandLinkButton_InitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    /// Allows for overriding the related default method
    pub fn OnInitStyleOption(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCommandLinkButton_OnInitStyleOption(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QCommandLinkButton_QBaseInitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    pub fn Event(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QCommandLinkButton_Event(@ptrCast(self), @ptrCast(e));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QCommandLinkButton_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QCommandLinkButton_QBaseEvent(@ptrCast(self), @ptrCast(e));
    }

    pub fn PaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCommandLinkButton_PaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnPaintEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCommandLinkButton_OnPaintEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QCommandLinkButton_QBasePaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCommandLinkButton_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcommandlinkbutton.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCommandLinkButton_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcommandlinkbutton.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCommandLinkButton_Delete(@ptrCast(self));
    }
};
